From 55d345f3c48615aee8ba9fed843ac0c4e9db5275 Mon Sep 17 00:00:00 2001
From: Simone Tellini <simone.tellini@robomagister.com>
Date: Thu, 5 Oct 2023 10:03:32 +0000
Subject: [PATCH] Fix u-boot default env for Engicam iMX8M Plus + C.Touch 2.0

---
 include/configs/imx8mp_icore.h | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/include/configs/imx8mp_icore.h b/include/configs/imx8mp_icore.h
index cfdfcfbd64..a00583141a 100644
--- a/include/configs/imx8mp_icore.h
+++ b/include/configs/imx8mp_icore.h
@@ -98,18 +98,20 @@
 	"boot_fdt=try\0" \
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fit=no\0" \
-	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
+	"fdtfile=imx8mp-icore-fasteth-ctouch2.dtb\0" \
 	"bootm_size=0x10000000\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
 	"mmcpart=1\0" \
-	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
+	"mmcblk=2\0" \
+	"mmcroot=/dev/mmcblk2p${mmcpart} rootwait rw\0" \
 	"mmcautodetect=yes\0" \
-	"mmcargs=setenv bootargs ${jh_clk} ${mcore_clk} console=${console} root=${mmcroot} " CMA_VALUE "\0 " \
-	"loadbootscript=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bsp_script};\0" \
+	"mmcargs=setenv bootargs ${jh_clk} ${mcore_clk} console=${console} root=/dev/mmcblk${mmcblk}p${mmcpart} " CMA_VALUE "\0 " \
+	"loadbootscript=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bsp_script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
-	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr_r} ${fdtfile}\0" \
+	"bootdir=/boot\0" \
+	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootdir}/${image}\0" \
+	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${bootdir}/${fdtfile}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"if test ${boot_fit} = yes || test ${boot_fit} = try; then " \
