From af6f2451752da52f1d4feabfe7103ac748f1f523 Mon Sep 17 00:00:00 2001
From: Simone Tellini <simone.tellini@robomagister.com>
Date: Fri, 6 Oct 2023 15:14:35 +0000
Subject: [PATCH] Fix mmcdev

---
 include/configs/imx8mp_icore.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/imx8mp_icore.h b/include/configs/imx8mp_icore.h
index a00583141a..abad11b116 100644
--- a/include/configs/imx8mp_icore.h
+++ b/include/configs/imx8mp_icore.h
@@ -100,7 +100,7 @@
 	"boot_fit=no\0" \
 	"fdtfile=imx8mp-icore-fasteth-ctouch2.dtb\0" \
 	"bootm_size=0x10000000\0" \
-	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
+	"mmcdev=2\0" \
 	"mmcpart=1\0" \
 	"mmcblk=2\0" \
 	"mmcroot=/dev/mmcblk2p${mmcpart} rootwait rw\0" \
